packages/microservices/client/client-proxy.ts

```diff
@@ -140,26 +141,32 @@ export abstract class ClientProxy {
   protected initializeSerializer(options: ClientOptions['options']) {
     this.serializer =
       (options &&
-        (options as
-          | RedisOptions['options']
-          | NatsOptions['options']
-          | MqttOptions['options']
-          | TcpClientOptions['options']
-          | RmqOptions['options']
-          | KafkaOptions['options']).serializer) ||
+        (
+          options as
+            | RedisOptions['options']
```

Comment:
what do you think about extracting these types into a Union type declared, something like : 
```suggestion
            const type Options = NatsOptions['options']
            | MqttOptions['options']
            | TcpClientOptions['options']
            | TcpTlsClientOptions['options']
            | RmqOptions['options']
            | KafkaOptions['options'];
```